src/main/java/org/intellij/sdk/codesync/PopulateBuffer.java

```diff
@@ -145,6 +146,8 @@ public static void populateBuffer(Project project) {
         if (!canRunDaemon) {
             return;
         }
+        S3FilesUploader.triggerS3Uploads(project);
```

Comment:
If i understood this correctly, it will be recalled for every iteration of populateBuffer, right? 
How are we making sure there are no duplicate instances? Different instances will have different value of `filesBeingProcessed` as per my understanding. 

**Case**
Previous s3 upload was in progress but daemon again called `S3FilesUploader.triggerS3Uploads(project);`